app/viewmodels/repository.js

```diff
@@ -1,11 +1,18 @@
-'use strict';
+﻿'use strict';
 define(['plugins/http', 'durandal/app', 'knockout', '../modules/github'], function (http, app, ko, Github) {
-    var github = new Github();
+    var github = Github.instance;
+
+    // defining these in the function scope as linkToDocument is not
+    // called in the right context, and this is cleaner than a bind in
+    // the template
+    var repositoryOwner =  ko.observable('');
+    var repositoryName = ko.observable('');
+    var commitSha = ko.observable('');
```

Comment:
If I recall correctly, the value of `this` in the invocation of `linkToDocument` was not the ViewModel, hence the need for these to exist in the variable scope. However, I could easily be mistaken, in which case this could be removed.